test/e2e/must_gather_operator_tests.go

```diff
@@ -1,114 +1,1186 @@
+				"SA should NOT be able to delete pods")
+
+			// Verify SA cannot create secrets
+			sar2 := &authorizationv1.SubjectAccessReview{
+				Spec: authorizationv1.SubjectAccessReviewSpec{
+					User: saUser,
+					ResourceAttributes: &authorizationv1.ResourceAttributes{
+						Verb:     "create",
+						Group:    "",
+						Resource: "secrets",
+					},
+				},
+			}
+			err = adminClient.Create(testCtx, sar2)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sar2.Status.Allowed).To(BeFalse(),
+				"SA should NOT be able to create secrets")
+		})
+	})
+
+	ginkgo.Context("Non-Admin User MustGather Collection Flow", func() {
+		var mustGatherName string
+		var createdMustGather *mustgatherv1alpha1.MustGather
+
+		ginkgo.BeforeEach(func() {
+			mustGatherName = fmt.Sprintf("test-non-admin-mg-%d", time.Now().Unix())
+		})
+
+		ginkgo.AfterEach(func() {
+			if createdMustGather != nil {
+				ginkgo.By("Cleaning up MustGather CR")
+				_ = adminClient.Delete(testCtx, createdMustGather)
+
+				// Wait for cleanup
+				Eventually(func() bool {
+					err := adminClient.Get(testCtx, client.ObjectKey{
+						Name:      mustGatherName,
+						Namespace: testNamespace,
+					}, &mustgatherv1alpha1.MustGather{})
+					return apierrors.IsNotFound(err)
+				}).WithTimeout(2 * time.Minute).WithPolling(5 * time.Second).Should(BeTrue())
+			}
+		})
+
+		ginkgo.It("non-admin user should successfully submit MustGather CR with SA", func() {
+			ginkgo.By("Non-admin user creating MustGather CR using impersonation")
+			createdMustGather = &mustgatherv1alpha1.MustGather{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      mustGatherName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						"test.description": "Non-admin user submitted MustGather",
+						"test.user":        nonAdminUser,
+					},
+				},
+				Spec: mustgatherv1alpha1.MustGatherSpec{
+					ServiceAccountName: ServiceAccount,
+					// Using default must-gather image
+				},
+			}
+
+			// Using impersonated non-admin client
+			err := nonAdminClient.Create(testCtx, createdMustGather)
+			Expect(err).NotTo(HaveOccurred(), "Non-admin user should be able to create MustGather CR")
+
+			ginkgo.By("Verifying MustGather CR was created successfully")
+			fetchedMG := &mustgatherv1alpha1.MustGather{}
+			err = nonAdminClient.Get(testCtx, client.ObjectKey{
+				Name:      mustGatherName,
+				Namespace: testNamespace,
+			}, fetchedMG)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(fetchedMG.Spec.ServiceAccountName).To(Equal(ServiceAccount))
+
+			ginkgo.GinkgoWriter.Printf("✅ Non-admin user '%s' successfully created MustGather CR: %s\n",
+				nonAdminUser, mustGatherName)
+		})
+
+		ginkgo.It("operator should create Job using ServiceAccount", func() {
+			ginkgo.By("Non-admin user creating MustGather CR")
+			createdMustGather = createMustGatherCR(mustGatherName, ServiceAccount)
+
+			ginkgo.By("Waiting for operator to create Job (non-admin user perspective)")
```

Comment:
```suggestion
			ginkgo.By("Waiting for operator to create Job")
```

Add "non-admin" in Ginkgo context and this is self-explanatory then.